[Linux][aarch64][NCE] Compilation error after #12237

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

Mainline 1654 or later

### Description of Issue

Compilation fails for Mainline 1654 or later, on Linux aarch64, where `HAS_NCE=1` (auto-detected).

### Expected Behavior

Compilation is successful, like with Mainline 1653.

### Reproduction Steps

0. On a machine running Fedora Asahi Remix 39
1. Using https://github.com/teohhanhui/rpms/tree/9802c68a2377aff99779ba7b629f35ee2120b648/yuzu
2. Build without modifications, observe that the build is successful.
3. Change `yuzu.spec`:

```patch
diff --git a/yuzu/yuzu.spec b/yuzu/yuzu.spec
index 0c2ca1b..67f7281 100644
--- a/yuzu/yuzu.spec
+++ b/yuzu/yuzu.spec
@@ -1,9 +1,9 @@
-%global date 20231216
-%global commit defda5dae4c504ffefda673903b8048543456fe1
+%global date 20231219
+%global commit af0eed53bbc326b43b4c6fe61f2b62d7a78de8ed
 %global shortcommit %(c=%{commit}; echo ${c:0:9})
 
 Name:           yuzu
-Version:        1653
+Version:        1656
 Release:        %autorelease
 Summary:        Nintendo Switch emulator
 
```
4. Build again and observe the compiler errors below.

### Log File

```
[1493/2078] /usr/lib64/ccache/g++ -DARCHITECTURE_arm64=1 -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DENABLE_WEB_SERVICE -DFMT_SHARED -DHAS_NCE=1 -DNDEBUG -DYUZU_UNIX=1 -D_FILE_OFFSET_BITS=64 -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/. -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./microprofile -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/glad/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/externals/SPIRV-Headers/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./stb -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./bc_decoder -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./mbedtls/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/oaknut/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/.. -isystem /usr/include/stb -isystem /usr/include/opus -isystem /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./renderdoc -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -std=gnu++20 -fwrapv -Werror=all -Werror=extra -Werror=missing-declarations -Werror=shadow -Werror=unused -Wno-attributes -Wno-invalid-offsetof -Wno-unused-parameter -Wno-array-bounds -Wno-stringop-overread -Wno-stringop-overflow -Werror=conversion -Wno-sign-conversion -Wno-cast-function-type -Winvalid-pch -include /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/src/core/CMakeFiles/core.dir/cmake_pch.hxx -MD -MT src/core/CMakeFiles/core.dir/arm/nce/arm_nce.cpp.o -MF src/core/CMakeFiles/core.dir/arm/nce/arm_nce.cpp.o.d -o src/core/CMakeFiles/core.dir/arm/nce/arm_nce.cpp.o -c /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/arm/nce/arm_nce.cpp
FAILED: src/core/CMakeFiles/core.dir/arm/nce/arm_nce.cpp.o 
/usr/lib64/ccache/g++ -DARCHITECTURE_arm64=1 -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DENABLE_WEB_SERVICE -DFMT_SHARED -DHAS_NCE=1 -DNDEBUG -DYUZU_UNIX=1 -D_FILE_OFFSET_BITS=64 -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/. -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./microprofile -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/glad/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/externals/SPIRV-Headers/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./stb -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./bc_decoder -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./mbedtls/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/oaknut/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/.. -isystem /usr/include/stb -isystem /usr/include/opus -isystem /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./renderdoc -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -std=gnu++20 -fwrapv -Werror=all -Werror=extra -Werror=missing-declarations -Werror=shadow -Werror=unused -Wno-attributes -Wno-invalid-offsetof -Wno-unused-parameter -Wno-array-bounds -Wno-stringop-overread -Wno-stringop-overflow -Werror=conversion -Wno-sign-conversion -Wno-cast-function-type -Winvalid-pch -include /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/src/core/CMakeFiles/core.dir/cmake_pch.hxx -MD -MT src/core/CMakeFiles/core.dir/arm/nce/arm_nce.cpp.o -MF src/core/CMakeFiles/core.dir/arm/nce/arm_nce.cpp.o.d -o src/core/CMakeFiles/core.dir/arm/nce/arm_nce.cpp.o -c /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/arm/nce/arm_nce.cpp
In file included from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/A64/a64_types.h:11,
                 from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./core/arm/nce/visitor_base.h:7,
                 from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./core/arm/nce/interpreter_visitor.h:10,
                 from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/arm/nce/arm_nce.cpp:9:
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include/mcl/assert.hpp:26: warning: "UNREACHABLE" redefined
   26 | #define UNREACHABLE() ASSERT_FALSE("Unreachable code!")
      | 
In file included from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./common/common_precompiled_headers.h:13,
                 from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/precompiled_headers.h:9,
                 from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/src/core/CMakeFiles/core.dir/cmake_pch.hxx:5,
                 from <command-line>:
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./common/assert.h:38: note: this is the location of the previous definition
   38 | #define UNREACHABLE()                                                                              \
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include/mcl/assert.hpp:28: warning: "ASSERT" redefined
   28 | #define ASSERT(expr)                                                     \
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./common/assert.h:22: note: this is the location of the previous definition
   22 | #define ASSERT(_a_)                                                                                \
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include/mcl/assert.hpp:41: warning: "ASSERT_MSG" redefined
   41 | #define ASSERT_MSG(expr, ...)                                                \
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./common/assert.h:30: note: this is the location of the previous definition
   30 | #define ASSERT_MSG(_a_, ...)                                                                       \
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include/mcl/assert.hpp:57: warning: "DEBUG_ASSERT" redefined
   57 | #    define DEBUG_ASSERT(expr) ASSUME(expr)
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./common/assert.h:54: note: this is the location of the previous definition
   54 | #define DEBUG_ASSERT(_a_)                                                                          \
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include/mcl/assert.hpp:58: warning: "DEBUG_ASSERT_MSG" redefined
   58 | #    define DEBUG_ASSERT_MSG(expr, ...) ASSUME(expr)
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./common/assert.h:57: note: this is the location of the previous definition
   57 | #define DEBUG_ASSERT_MSG(_a_, _desc_, ...)                                                         \
      | 
In file included from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./core/arm/nce/visitor_base.h:8:
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/imm.h: In constructor ‘Dynarmic::Imm<bit_size_>::Imm(u32)’:
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/imm.h:28:22: error: declaration of ‘value’ shadows a member of ‘Dynarmic::Imm<bit_size_>’ [-Werror=shadow]
   28 |     explicit Imm(u32 value)
      |                  ~~~~^~~~~
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/imm.h:86:9: note: shadowed declaration is here
   86 |     u32 value;
      |         ^~~~~
cc1plus: some warnings being treated as errors
[1494/2078] /usr/lib64/ccache/g++ -DARCHITECTURE_arm64=1 -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DENABLE_WEB_SERVICE -DFMT_SHARED -DHAS_NCE=1 -DNDEBUG -DYUZU_UNIX=1 -D_FILE_OFFSET_BITS=64 -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/. -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./microprofile -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/glad/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/externals/SPIRV-Headers/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./stb -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./bc_decoder -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./mbedtls/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/oaknut/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/.. -isystem /usr/include/stb -isystem /usr/include/opus -isystem /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./renderdoc -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -std=gnu++20 -fwrapv -Werror=all -Werror=extra -Werror=missing-declarations -Werror=shadow -Werror=unused -Wno-attributes -Wno-invalid-offsetof -Wno-unused-parameter -Wno-array-bounds -Wno-stringop-overread -Wno-stringop-overflow -Werror=conversion -Wno-sign-conversion -Wno-cast-function-type -Winvalid-pch -include /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/src/core/CMakeFiles/core.dir/cmake_pch.hxx -MD -MT src/core/CMakeFiles/core.dir/perf_stats.cpp.o -MF src/core/CMakeFiles/core.dir/perf_stats.cpp.o.d -o src/core/CMakeFiles/core.dir/perf_stats.cpp.o -c /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/perf_stats.cpp
[1495/2078] /usr/lib64/ccache/g++ -DARCHITECTURE_arm64=1 -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DENABLE_WEB_SERVICE -DFMT_SHARED -DHAS_NCE=1 -DNDEBUG -DYUZU_UNIX=1 -D_FILE_OFFSET_BITS=64 -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/. -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./microprofile -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/glad/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/externals/SPIRV-Headers/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./stb -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./bc_decoder -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./mbedtls/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/oaknut/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/.. -isystem /usr/include/stb -isystem /usr/include/opus -isystem /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./renderdoc -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -std=gnu++20 -fwrapv -Werror=all -Werror=extra -Werror=missing-declarations -Werror=shadow -Werror=unused -Wno-attributes -Wno-invalid-offsetof -Wno-unused-parameter -Wno-array-bounds -Wno-stringop-overread -Wno-stringop-overflow -Werror=conversion -Wno-sign-conversion -Wno-cast-function-type -Winvalid-pch -include /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/src/core/CMakeFiles/core.dir/cmake_pch.hxx -MD -MT src/core/CMakeFiles/core.dir/arm/dynarmic/dynarmic_cp15.cpp.o -MF src/core/CMakeFiles/core.dir/arm/dynarmic/dynarmic_cp15.cpp.o.d -o src/core/CMakeFiles/core.dir/arm/dynarmic/dynarmic_cp15.cpp.o -c /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/arm/dynarmic/dynarmic_cp15.cpp
[1496/2078] /usr/lib64/ccache/g++ -DARCHITECTURE_arm64=1 -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DENABLE_WEB_SERVICE -DFMT_SHARED -DHAS_NCE=1 -DNDEBUG -DYUZU_UNIX=1 -D_FILE_OFFSET_BITS=64 -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/. -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./microprofile -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/glad/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/externals/SPIRV-Headers/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./stb -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./bc_decoder -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./mbedtls/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/oaknut/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/.. -isystem /usr/include/stb -isystem /usr/include/opus -isystem /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./renderdoc -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -std=gnu++20 -fwrapv -Werror=all -Werror=extra -Werror=missing-declarations -Werror=shadow -Werror=unused -Wno-attributes -Wno-invalid-offsetof -Wno-unused-parameter -Wno-array-bounds -Wno-stringop-overread -Wno-stringop-overflow -Werror=conversion -Wno-sign-conversion -Wno-cast-function-type -Winvalid-pch -include /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/src/core/CMakeFiles/core.dir/cmake_pch.hxx -MD -MT src/core/CMakeFiles/core.dir/arm/dynarmic/arm_dynarmic_64.cpp.o -MF src/core/CMakeFiles/core.dir/arm/dynarmic/arm_dynarmic_64.cpp.o.d -o src/core/CMakeFiles/core.dir/arm/dynarmic/arm_dynarmic_64.cpp.o -c /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/arm/dynarmic/arm_dynarmic_64.cpp
[1497/2078] /usr/lib64/ccache/g++ -DARCHITECTURE_arm64=1 -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DENABLE_WEB_SERVICE -DFMT_SHARED -DHAS_NCE=1 -DNDEBUG -DYUZU_UNIX=1 -D_FILE_OFFSET_BITS=64 -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/. -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./microprofile -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/glad/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/externals/SPIRV-Headers/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./stb -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./bc_decoder -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./mbedtls/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/oaknut/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/.. -isystem /usr/include/stb -isystem /usr/include/opus -isystem /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./renderdoc -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -std=gnu++20 -fwrapv -Werror=all -Werror=extra -Werror=missing-declarations -Werror=shadow -Werror=unused -Wno-attributes -Wno-invalid-offsetof -Wno-unused-parameter -Wno-array-bounds -Wno-stringop-overread -Wno-stringop-overflow -Werror=conversion -Wno-sign-conversion -Wno-cast-function-type -Winvalid-pch -include /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/src/core/CMakeFiles/core.dir/cmake_pch.hxx -MD -MT src/core/CMakeFiles/core.dir/arm/dynarmic/arm_dynarmic_32.cpp.o -MF src/core/CMakeFiles/core.dir/arm/dynarmic/arm_dynarmic_32.cpp.o.d -o src/core/CMakeFiles/core.dir/arm/dynarmic/arm_dynarmic_32.cpp.o -c /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/arm/dynarmic/arm_dynarmic_32.cpp
[1498/2078] /usr/lib64/ccache/g++ -DARCHITECTURE_arm64=1 -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DENABLE_WEB_SERVICE -DFMT_SHARED -DHAS_NCE=1 -DNDEBUG -DYUZU_UNIX=1 -D_FILE_OFFSET_BITS=64 -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/. -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./microprofile -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/glad/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/externals/SPIRV-Headers/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./stb -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./bc_decoder -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./mbedtls/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/oaknut/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/.. -isystem /usr/include/stb -isystem /usr/include/opus -isystem /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./renderdoc -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -std=gnu++20 -fwrapv -Werror=all -Werror=extra -Werror=missing-declarations -Werror=shadow -Werror=unused -Wno-attributes -Wno-invalid-offsetof -Wno-unused-parameter -Wno-array-bounds -Wno-stringop-overread -Wno-stringop-overflow -Werror=conversion -Wno-sign-conversion -Wno-cast-function-type -Winvalid-pch -include /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/src/core/CMakeFiles/core.dir/cmake_pch.hxx -MD -MT src/core/CMakeFiles/core.dir/arm/nce/patcher.cpp.o -MF src/core/CMakeFiles/core.dir/arm/nce/patcher.cpp.o.d -o src/core/CMakeFiles/core.dir/arm/nce/patcher.cpp.o -c /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/arm/nce/patcher.cpp
[1499/2078] /usr/lib64/ccache/g++ -DARCHITECTURE_arm64=1 -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DENABLE_WEB_SERVICE -DFMT_SHARED -DHAS_NCE=1 -DNDEBUG -DYUZU_UNIX=1 -D_FILE_OFFSET_BITS=64 -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/. -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./microprofile -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/glad/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/externals/SPIRV-Headers/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./stb -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./bc_decoder -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./mbedtls/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/oaknut/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/.. -isystem /usr/include/stb -isystem /usr/include/opus -isystem /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./renderdoc -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -std=gnu++20 -fwrapv -Werror=all -Werror=extra -Werror=missing-declarations -Werror=shadow -Werror=unused -Wno-attributes -Wno-invalid-offsetof -Wno-unused-parameter -Wno-array-bounds -Wno-stringop-overread -Wno-stringop-overflow -Werror=conversion -Wno-sign-conversion -Wno-cast-function-type -Winvalid-pch -include /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/src/core/CMakeFiles/core.dir/cmake_pch.hxx -MD -MT src/core/CMakeFiles/core.dir/reporter.cpp.o -MF src/core/CMakeFiles/core.dir/reporter.cpp.o.d -o src/core/CMakeFiles/core.dir/reporter.cpp.o -c /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/reporter.cpp
[1500/2078] /usr/lib64/ccache/g++ -DARCHITECTURE_arm64=1 -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DENABLE_WEB_SERVICE -DFMT_SHARED -DHAS_NCE=1 -DNDEBUG -DYUZU_UNIX=1 -D_FILE_OFFSET_BITS=64 -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/. -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./microprofile -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/glad/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/externals/SPIRV-Headers/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./stb -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./bc_decoder -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./mbedtls/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/oaknut/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/.. -isystem /usr/include/stb -isystem /usr/include/opus -isystem /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./renderdoc -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -std=gnu++20 -fwrapv -Werror=all -Werror=extra -Werror=missing-declarations -Werror=shadow -Werror=unused -Wno-attributes -Wno-invalid-offsetof -Wno-unused-parameter -Wno-array-bounds -Wno-stringop-overread -Wno-stringop-overflow -Werror=conversion -Wno-sign-conversion -Wno-cast-function-type -Winvalid-pch -include /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/src/core/CMakeFiles/core.dir/cmake_pch.hxx -MD -MT src/core/CMakeFiles/core.dir/arm/nce/interpreter_visitor.cpp.o -MF src/core/CMakeFiles/core.dir/arm/nce/interpreter_visitor.cpp.o.d -o src/core/CMakeFiles/core.dir/arm/nce/interpreter_visitor.cpp.o -c /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/arm/nce/interpreter_visitor.cpp
FAILED: src/core/CMakeFiles/core.dir/arm/nce/interpreter_visitor.cpp.o 
/usr/lib64/ccache/g++ -DARCHITECTURE_arm64=1 -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DENABLE_WEB_SERVICE -DFMT_SHARED -DHAS_NCE=1 -DNDEBUG -DYUZU_UNIX=1 -D_FILE_OFFSET_BITS=64 -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/. -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./microprofile -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/glad/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/externals/SPIRV-Headers/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./stb -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./bc_decoder -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./mbedtls/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/oaknut/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/.. -isystem /usr/include/stb -isystem /usr/include/opus -isystem /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./renderdoc -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -std=gnu++20 -fwrapv -Werror=all -Werror=extra -Werror=missing-declarations -Werror=shadow -Werror=unused -Wno-attributes -Wno-invalid-offsetof -Wno-unused-parameter -Wno-array-bounds -Wno-stringop-overread -Wno-stringop-overflow -Werror=conversion -Wno-sign-conversion -Wno-cast-function-type -Winvalid-pch -include /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/src/core/CMakeFiles/core.dir/cmake_pch.hxx -MD -MT src/core/CMakeFiles/core.dir/arm/nce/interpreter_visitor.cpp.o -MF src/core/CMakeFiles/core.dir/arm/nce/interpreter_visitor.cpp.o.d -o src/core/CMakeFiles/core.dir/arm/nce/interpreter_visitor.cpp.o -c /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/arm/nce/interpreter_visitor.cpp
In file included from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/A64/a64_types.h:11,
                 from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./core/arm/nce/visitor_base.h:7,
                 from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./core/arm/nce/interpreter_visitor.h:10,
                 from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/arm/nce/interpreter_visitor.cpp:6:
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include/mcl/assert.hpp:26: warning: "UNREACHABLE" redefined
   26 | #define UNREACHABLE() ASSERT_FALSE("Unreachable code!")
      | 
In file included from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./common/common_precompiled_headers.h:13,
                 from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/precompiled_headers.h:9,
                 from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/src/core/CMakeFiles/core.dir/cmake_pch.hxx:5,
                 from <command-line>:
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./common/assert.h:38: note: this is the location of the previous definition
   38 | #define UNREACHABLE()                                                                              \
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include/mcl/assert.hpp:28: warning: "ASSERT" redefined
   28 | #define ASSERT(expr)                                                     \
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./common/assert.h:22: note: this is the location of the previous definition
   22 | #define ASSERT(_a_)                                                                                \
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include/mcl/assert.hpp:41: warning: "ASSERT_MSG" redefined
   41 | #define ASSERT_MSG(expr, ...)                                                \
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./common/assert.h:30: note: this is the location of the previous definition
   30 | #define ASSERT_MSG(_a_, ...)                                                                       \
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include/mcl/assert.hpp:57: warning: "DEBUG_ASSERT" redefined
   57 | #    define DEBUG_ASSERT(expr) ASSUME(expr)
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./common/assert.h:54: note: this is the location of the previous definition
   54 | #define DEBUG_ASSERT(_a_)                                                                          \
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include/mcl/assert.hpp:58: warning: "DEBUG_ASSERT_MSG" redefined
   58 | #    define DEBUG_ASSERT_MSG(expr, ...) ASSUME(expr)
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./common/assert.h:57: note: this is the location of the previous definition
   57 | #define DEBUG_ASSERT_MSG(_a_, _desc_, ...)                                                         \
      | 
In file included from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./core/arm/nce/visitor_base.h:8:
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/imm.h: In constructor ‘Dynarmic::Imm<bit_size_>::Imm(u32)’:
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/imm.h:28:22: error: declaration of ‘value’ shadows a member of ‘Dynarmic::Imm<bit_size_>’ [-Werror=shadow]
   28 |     explicit Imm(u32 value)
      |                  ~~~~^~~~~
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/imm.h:86:9: note: shadowed declaration is here
   86 |     u32 value;
      |         ^~~~~
In file included from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/A64/decoder/a64.h:19,
                 from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/arm/nce/interpreter_visitor.cpp:8:
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/decoder/matcher.h: In constructor ‘Dynarmic::Decoder::Matcher<Visitor, OpcodeType>::Matcher(const char*, opcode_type, opcode_type, handler_function)’:
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/decoder/matcher.h:32:67: error: declaration of ‘expected’ shadows a member of ‘Dynarmic::Decoder::Matcher<Visitor, OpcodeType>’ [-Werror=shadow]
   32 |     Matcher(const char* const name, opcode_type mask, opcode_type expected, handler_function func)
      |                                                       ~~~~~~~~~~~~^~~~~~~~
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/decoder/matcher.h:72:17: note: shadowed declaration is here
   72 |     opcode_type expected;
      |                 ^~~~~~~~
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/decoder/matcher.h:32:49: error: declaration of ‘mask’ shadows a member of ‘Dynarmic::Decoder::Matcher<Visitor, OpcodeType>’ [-Werror=shadow]
   32 |     Matcher(const char* const name, opcode_type mask, opcode_type expected, handler_function func)
      |                                     ~~~~~~~~~~~~^~~~
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/decoder/matcher.h:71:17: note: shadowed declaration is here
   71 |     opcode_type mask;
      |                 ^~~~
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/decoder/matcher.h:32:31: error: declaration of ‘name’ shadows a member of ‘Dynarmic::Decoder::Matcher<Visitor, OpcodeType>’ [-Werror=shadow]
   32 |     Matcher(const char* const name, opcode_type mask, opcode_type expected, handler_function func)
      |             ~~~~~~~~~~~~~~~~~~^~~~
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/decoder/matcher.h:70:17: note: shadowed declaration is here
   70 |     const char* name;
      |                 ^~~~
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/arm/nce/interpreter_visitor.cpp: In function ‘std::optional<long unsigned int> Core::MatchAndExecuteOneInstruction(Memory::Memory&, mcontext_t*, fpsimd_context*)’:
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/arm/nce/interpreter_visitor.cpp:802:16: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  802 |     u64& sp = *reinterpret_cast<u64*>(&context->sp);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/arm/nce/interpreter_visitor.cpp:803:22: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  803 |     const u64& pc = *reinterpret_cast<u64*>(&context->pc);
      |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: some warnings being treated as errors
ninja: build stopped: subcommand failed.
```

### System Configuration

CPU: Apple M2
GPU/Driver: Apple M2 / Asahi
RAM: 16G
OS: Fedora Asahi Remix 39

Looks like it made no difference:

```
[1578/2078] /usr/lib64/ccache/g++ -DARCHITECTURE_arm64=1 -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DENABLE_WEB_SERVICE -DFMT_SHARED -DHAS_NCE=1 -DNDEBUG -DYUZU_UNIX=1 -D_FILE_OFFSET_BITS=64 -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/. -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./microprofile -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/glad/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/externals/SPIRV-Headers/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./stb -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./bc_decoder -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./mbedtls/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/oaknut/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/.. -isystem /usr/include/stb -isystem /usr/include/opus -isystem /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./renderdoc -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -std=gnu++20 -fwrapv -Werror=all -Werror=extra -Werror=missing-declarations -Werror=shadow -Werror=unused -Wno-attributes -Wno-invalid-offsetof -Wno-unused-parameter -Wno-array-bounds -Wno-stringop-overread -Wno-stringop-overflow -Werror=conversion -Wno-sign-conversion -Wno-cast-function-type -Winvalid-pch -include /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/src/core/CMakeFiles/core.dir/cmake_pch.hxx -MD -MT src/core/CMakeFiles/core.dir/arm/nce/interpreter_visitor.cpp.o -MF src/core/CMakeFiles/core.dir/arm/nce/interpreter_visitor.cpp.o.d -o src/core/CMakeFiles/core.dir/arm/nce/interpreter_visitor.cpp.o -c /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/arm/nce/interpreter_visitor.cpp
FAILED: src/core/CMakeFiles/core.dir/arm/nce/interpreter_visitor.cpp.o 
/usr/lib64/ccache/g++ -DARCHITECTURE_arm64=1 -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DENABLE_WEB_SERVICE -DFMT_SHARED -DHAS_NCE=1 -DNDEBUG -DYUZU_UNIX=1 -D_FILE_OFFSET_BITS=64 -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/. -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./microprofile -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/glad/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/sirit/externals/SPIRV-Headers/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./stb -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./bc_decoder -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/externals/nx_tzdb/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./mbedtls/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/oaknut/include -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/.. -isystem /usr/include/stb -isystem /usr/include/opus -isystem /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./renderdoc -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -std=gnu++20 -fwrapv -Werror=all -Werror=extra -Werror=missing-declarations -Werror=shadow -Werror=unused -Wno-attributes -Wno-invalid-offsetof -Wno-unused-parameter -Wno-array-bounds -Wno-stringop-overread -Wno-stringop-overflow -Werror=conversion -Wno-sign-conversion -Wno-cast-function-type -Winvalid-pch -include /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/src/core/CMakeFiles/core.dir/cmake_pch.hxx -MD -MT src/core/CMakeFiles/core.dir/arm/nce/interpreter_visitor.cpp.o -MF src/core/CMakeFiles/core.dir/arm/nce/interpreter_visitor.cpp.o.d -o src/core/CMakeFiles/core.dir/arm/nce/interpreter_visitor.cpp.o -c /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/arm/nce/interpreter_visitor.cpp
In file included from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/A64/a64_types.h:11,
                 from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./core/arm/nce/visitor_base.h:10,
                 from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./core/arm/nce/interpreter_visitor.h:10,
                 from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/arm/nce/interpreter_visitor.cpp:6:
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include/mcl/assert.hpp:26: warning: "UNREACHABLE" redefined
   26 | #define UNREACHABLE() ASSERT_FALSE("Unreachable code!")
      | 
In file included from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./common/common_precompiled_headers.h:13,
                 from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/precompiled_headers.h:9,
                 from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/src/core/CMakeFiles/core.dir/cmake_pch.hxx:5,
                 from <command-line>:
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./common/assert.h:38: note: this is the location of the previous definition
   38 | #define UNREACHABLE()                                                                              \
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include/mcl/assert.hpp:28: warning: "ASSERT" redefined
   28 | #define ASSERT(expr)                                                     \
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./common/assert.h:22: note: this is the location of the previous definition
   22 | #define ASSERT(_a_)                                                                                \
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include/mcl/assert.hpp:41: warning: "ASSERT_MSG" redefined
   41 | #define ASSERT_MSG(expr, ...)                                                \
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./common/assert.h:30: note: this is the location of the previous definition
   30 | #define ASSERT_MSG(_a_, ...)                                                                       \
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include/mcl/assert.hpp:57: warning: "DEBUG_ASSERT" redefined
   57 | #    define DEBUG_ASSERT(expr) ASSUME(expr)
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./common/assert.h:54: note: this is the location of the previous definition
   54 | #define DEBUG_ASSERT(_a_)                                                                          \
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/externals/mcl/src/../include/mcl/assert.hpp:58: warning: "DEBUG_ASSERT_MSG" redefined
   58 | #    define DEBUG_ASSERT_MSG(expr, ...) ASSUME(expr)
      | 
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/./common/assert.h:57: note: this is the location of the previous definition
   57 | #define DEBUG_ASSERT_MSG(_a_, _desc_, ...)                                                         \
      | 
In file included from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/A64/decoder/a64.h:19,
                 from /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/arm/nce/interpreter_visitor.cpp:8:
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/decoder/matcher.h: In constructor ‘Dynarmic::Decoder::Matcher<Visitor, OpcodeType>::Matcher(const char*, opcode_type, opcode_type, handler_function)’:
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/decoder/matcher.h:32:67: error: declaration of ‘expected’ shadows a member of ‘Dynarmic::Decoder::Matcher<Visitor, OpcodeType>’ [-Werror=shadow]
   32 |     Matcher(const char* const name, opcode_type mask, opcode_type expected, handler_function func)
      |                                                       ~~~~~~~~~~~~^~~~~~~~
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/decoder/matcher.h:72:17: note: shadowed declaration is here
   72 |     opcode_type expected;
      |                 ^~~~~~~~
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/decoder/matcher.h:32:49: error: declaration of ‘mask’ shadows a member of ‘Dynarmic::Decoder::Matcher<Visitor, OpcodeType>’ [-Werror=shadow]
   32 |     Matcher(const char* const name, opcode_type mask, opcode_type expected, handler_function func)
      |                                     ~~~~~~~~~~~~^~~~
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/decoder/matcher.h:71:17: note: shadowed declaration is here
   71 |     opcode_type mask;
      |                 ^~~~
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/decoder/matcher.h:32:31: error: declaration of ‘name’ shadows a member of ‘Dynarmic::Decoder::Matcher<Visitor, OpcodeType>’ [-Werror=shadow]
   32 |     Matcher(const char* const name, opcode_type mask, opcode_type expected, handler_function func)
      |             ~~~~~~~~~~~~~~~~~~^~~~
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/../dynarmic/frontend/decoder/matcher.h:70:17: note: shadowed declaration is here
   70 |     const char* name;
      |                 ^~~~
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/arm/nce/interpreter_visitor.cpp: In function ‘std::optional<long unsigned int> Core::MatchAndExecuteOneInstruction(Memory::Memory&, mcontext_t*, fpsimd_context*)’:
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/arm/nce/interpreter_visitor.cpp:802:16: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  802 |     u64& sp = *reinterpret_cast<u64*>(&context->sp);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/core/arm/nce/interpreter_visitor.cpp:803:22: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  803 |     const u64& pc = *reinterpret_cast<u64*>(&context->pc);
      |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: some warnings being treated as errors
[1579/2078] /usr/lib64/ccache/g++ -DARCHITECTURE_arm64=1 -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DFMT_SHARED -DHAS_NCE=1 -DHAVE_CUBEB=1 -DHAVE_SDL2 -DNDEBUG -DYUZU_UNIX=1 -D_FILE_OFFSET_BITS=64 -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/. -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./microprofile -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/.. -isystem /usr/include/stb -isystem /usr/include/opus -isystem /usr/include/SDL2 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -std=gnu++20 -fwrapv -Werror=all -Werror=extra -Werror=missing-declarations -Werror=shadow -Werror=unused -Wno-attributes -Wno-invalid-offsetof -Wno-unused-parameter -Wno-array-bounds -Wno-stringop-overread -Wno-stringop-overflow -Werror=conversion -Wno-sign-conversion -Winvalid-pch -include /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/src/audio_core/CMakeFiles/audio_core.dir/cmake_pch.hxx -MD -MT src/audio_core/CMakeFiles/audio_core.dir/renderer/voice/voice_info.cpp.o -MF src/audio_core/CMakeFiles/audio_core.dir/renderer/voice/voice_info.cpp.o.d -o src/audio_core/CMakeFiles/audio_core.dir/renderer/voice/voice_info.cpp.o -c /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/audio_core/renderer/voice/voice_info.cpp
[1580/2078] /usr/lib64/ccache/g++ -DARCHITECTURE_arm64=1 -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DFMT_SHARED -DHAS_NCE=1 -DHAVE_CUBEB=1 -DHAVE_SDL2 -DNDEBUG -DYUZU_UNIX=1 -D_FILE_OFFSET_BITS=64 -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/. -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./microprofile -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/.. -isystem /usr/include/stb -isystem /usr/include/opus -isystem /usr/include/SDL2 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -std=gnu++20 -fwrapv -Werror=all -Werror=extra -Werror=missing-declarations -Werror=shadow -Werror=unused -Wno-attributes -Wno-invalid-offsetof -Wno-unused-parameter -Wno-array-bounds -Wno-stringop-overread -Wno-stringop-overflow -Werror=conversion -Wno-sign-conversion -Winvalid-pch -include /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/src/audio_core/CMakeFiles/audio_core.dir/cmake_pch.hxx -MD -MT src/audio_core/CMakeFiles/audio_core.dir/renderer/voice/voice_context.cpp.o -MF src/audio_core/CMakeFiles/audio_core.dir/renderer/voice/voice_context.cpp.o.d -o src/audio_core/CMakeFiles/audio_core.dir/renderer/voice/voice_context.cpp.o -c /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/audio_core/renderer/voice/voice_context.cpp
[1581/2078] /usr/lib64/ccache/g++ -DARCHITECTURE_arm64=1 -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DFMT_SHARED -DHAS_NCE=1 -DHAVE_CUBEB=1 -DHAVE_SDL2 -DNDEBUG -DYUZU_UNIX=1 -D_FILE_OFFSET_BITS=64 -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/. -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./microprofile -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/.. -isystem /usr/include/stb -isystem /usr/include/opus -isystem /usr/include/SDL2 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -std=gnu++20 -fwrapv -Werror=all -Werror=extra -Werror=missing-declarations -Werror=shadow -Werror=unused -Wno-attributes -Wno-invalid-offsetof -Wno-unused-parameter -Wno-array-bounds -Wno-stringop-overread -Wno-stringop-overflow -Werror=conversion -Wno-sign-conversion -Winvalid-pch -include /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/src/audio_core/CMakeFiles/audio_core.dir/cmake_pch.hxx -MD -MT src/audio_core/CMakeFiles/audio_core.dir/sink/sink_details.cpp.o -MF src/audio_core/CMakeFiles/audio_core.dir/sink/sink_details.cpp.o.d -o src/audio_core/CMakeFiles/audio_core.dir/sink/sink_details.cpp.o -c /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/audio_core/sink/sink_details.cpp
[1582/2078] /usr/lib64/ccache/g++ -DARCHITECTURE_arm64=1 -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DFMT_SHARED -DHAS_NCE=1 -DHAVE_CUBEB=1 -DHAVE_SDL2 -DNDEBUG -DYUZU_UNIX=1 -D_FILE_OFFSET_BITS=64 -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/. -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./microprofile -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/.. -isystem /usr/include/stb -isystem /usr/include/opus -isystem /usr/include/SDL2 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -std=gnu++20 -fwrapv -Werror=all -Werror=extra -Werror=missing-declarations -Werror=shadow -Werror=unused -Wno-attributes -Wno-invalid-offsetof -Wno-unused-parameter -Wno-array-bounds -Wno-stringop-overread -Wno-stringop-overflow -Werror=conversion -Wno-sign-conversion -Winvalid-pch -include /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/src/audio_core/CMakeFiles/audio_core.dir/cmake_pch.hxx -MD -MT src/audio_core/CMakeFiles/audio_core.dir/renderer/system_manager.cpp.o -MF src/audio_core/CMakeFiles/audio_core.dir/renderer/system_manager.cpp.o.d -o src/audio_core/CMakeFiles/audio_core.dir/renderer/system_manager.cpp.o -c /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/audio_core/renderer/system_manager.cpp
[1583/2078] /usr/lib64/ccache/g++ -DARCHITECTURE_arm64=1 -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DFMT_SHARED -DHAS_NCE=1 -DHAVE_CUBEB=1 -DHAVE_SDL2 -DNDEBUG -DYUZU_UNIX=1 -D_FILE_OFFSET_BITS=64 -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/. -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./microprofile -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/.. -isystem /usr/include/stb -isystem /usr/include/opus -isystem /usr/include/SDL2 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -std=gnu++20 -fwrapv -Werror=all -Werror=extra -Werror=missing-declarations -Werror=shadow -Werror=unused -Wno-attributes -Wno-invalid-offsetof -Wno-unused-parameter -Wno-array-bounds -Wno-stringop-overread -Wno-stringop-overflow -Werror=conversion -Wno-sign-conversion -Winvalid-pch -include /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/src/audio_core/CMakeFiles/audio_core.dir/cmake_pch.hxx -MD -MT src/audio_core/CMakeFiles/audio_core.dir/renderer/system.cpp.o -MF src/audio_core/CMakeFiles/audio_core.dir/renderer/system.cpp.o.d -o src/audio_core/CMakeFiles/audio_core.dir/renderer/system.cpp.o -c /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/audio_core/renderer/system.cpp
[1584/2078] /usr/lib64/ccache/g++ -DARCHITECTURE_arm64=1 -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DFMT_SHARED -DHAS_NCE=1 -DHAVE_CUBEB=1 -DHAVE_SDL2 -DNDEBUG -DYUZU_UNIX=1 -D_FILE_OFFSET_BITS=64 -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/. -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./microprofile -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/.. -isystem /usr/include/stb -isystem /usr/include/opus -isystem /usr/include/SDL2 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -std=gnu++20 -fwrapv -Werror=all -Werror=extra -Werror=missing-declarations -Werror=shadow -Werror=unused -Wno-attributes -Wno-invalid-offsetof -Wno-unused-parameter -Wno-array-bounds -Wno-stringop-overread -Wno-stringop-overflow -Werror=conversion -Wno-sign-conversion -Winvalid-pch -include /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/src/audio_core/CMakeFiles/audio_core.dir/cmake_pch.hxx -MD -MT src/audio_core/CMakeFiles/audio_core.dir/sink/cubeb_sink.cpp.o -MF src/audio_core/CMakeFiles/audio_core.dir/sink/cubeb_sink.cpp.o.d -o src/audio_core/CMakeFiles/audio_core.dir/sink/cubeb_sink.cpp.o -c /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/audio_core/sink/cubeb_sink.cpp
[1585/2078] /usr/lib64/ccache/g++ -DARCHITECTURE_arm64=1 -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DFMT_SHARED -DHAS_NCE=1 -DHAVE_CUBEB=1 -DHAVE_SDL2 -DNDEBUG -DYUZU_UNIX=1 -D_FILE_OFFSET_BITS=64 -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/. -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/./microprofile -I/builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/externals/dynarmic/src/dynarmic/.. -isystem /usr/include/stb -isystem /usr/include/opus -isystem /usr/include/SDL2 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -std=gnu++20 -fwrapv -Werror=all -Werror=extra -Werror=missing-declarations -Werror=shadow -Werror=unused -Wno-attributes -Wno-invalid-offsetof -Wno-unused-parameter -Wno-array-bounds -Wno-stringop-overread -Wno-stringop-overflow -Werror=conversion -Wno-sign-conversion -Winvalid-pch -include /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/redhat-linux-build/src/audio_core/CMakeFiles/audio_core.dir/cmake_pch.hxx -MD -MT src/audio_core/CMakeFiles/audio_core.dir/sink/sink_stream.cpp.o -MF src/audio_core/CMakeFiles/audio_core.dir/sink/sink_stream.cpp.o.d -o src/audio_core/CMakeFiles/audio_core.dir/sink/sink_stream.cpp.o -c /builddir/build/BUILD/yuzu-linux-20231219-af0eed53b-source/src/audio_core/sink/sink_stream.cpp
ninja: build stopped: subcommand failed.
```

--- a/src/core/arm/nce/interpreter_visitor.cpp
+++ b/src/core/arm/nce/interpreter_visitor.cpp
@@ -5,8 +5,6 @@
 #include "common/bit_cast.h"
 #include "core/arm/nce/interpreter_visitor.h"
 
-#include <dynarmic/frontend/A64/decoder/a64.h>
-
 namespace Core {
 
 template <u32 BitSize>

--- a/src/core/arm/nce/visitor_base.h
+++ b/src/core/arm/nce/visitor_base.h
@@ -4,8 +4,14 @@
 
 #pragma once
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wshadow"
+
 #include <dynarmic/frontend/A64/a64_types.h>
+#include <dynarmic/frontend/A64/decoder/a64.h>
 #include <dynarmic/frontend/imm.h>
+
+#pragma GCC diagnostic pop
 
 namespace Core {
 

